apps/game_backend/lib/game_backend/units.ex

```diff
@@ -131,15 +131,18 @@ defmodule GameBackend.Units do
   Get a user's unit associated to the given character.
   Fails if there are more than one unit of the same character. Returns nil if there are none.
   """
-  def get_unit_by_character_name(character_name, user_id),
-    do:
-      Repo.one(
-        from(unit in user_units_query(user_id),
-          join: character in Character,
-          on: unit.character_id == character.id,
-          where: character.name == ^character_name
-        )
-      )
+  def get_unit_by_character_name(character_name, user_id) do
+    case Repo.one(
+           from(unit in user_units_query(user_id),
```

Comment:
I know every query has this but isn't it easier to just have `where: unit.user_id == ^user_id` in the query instead of using these for the base query?

Reply:
I think it shortens the code and when you know what it does its easier to follow the rest of the query.